apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -190,14 +198,32 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
 
       // We don't use JsonFile/jju here because shrinkwrap.json is a special NPM file format
       // and typically very large, so we want to load it the same way that NPM does.
-      const parsedData: IPnpmShrinkwrapYaml = yaml.safeLoad(FileSystem.readFile(shrinkwrapYamlFilename).toString());
+      const shrinkwrapContent: string = FileSystem.readFile(shrinkwrapYamlFilename).toString();
+      const parsedData: IPnpmShrinkwrapYaml = yaml.safeLoad(shrinkwrapContent);
 
-      return new PnpmShrinkwrapFile(parsedData, shrinkwrapYamlFilename);
+      // Grab the shrinkwrap hash out of the comment where we store it.
+      const hashIndex: number = shrinkwrapContent.lastIndexOf(PnpmShrinkwrapFile._shrinkwrapHashPrefix);
```

Comment:
It's probably safe to assume this is on the last non-blank line.

Reply:
I agree. However since it's a string I'm looking at, I still need to do the index to figure out where that last blank line begins. Given that, I just look backwards for the comment